tests/test_scatterwidgets.py

```diff
@@ -1,58 +1,404 @@
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 1
+
+    # Simulate double click
+    view_widget = plot_widget.scatter_plot.scene().views()[0].viewport()
+
+    point1 = [0.5, 0.5]
+    viewport_pos1 = plot_coords_to_viewport(plot_widget, *point1)
+
+    qtbot.mouseDClick(view_widget, QtCore.Qt.LeftButton, pos=viewport_pos1)
+    qtbot.wait(50)
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_remove_hovered_roi(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi = pg.RectROI([0, 0], [1, 1])
+    plot_widget.roi_list = [roi]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 1
+
+    # Simulate hovering over the ROI
+    roi.mouseHovering = True
+
+    plot_widget.remove_hovered_roi()
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_remove_all_rois(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi1 = pg.RectROI([0, 0], [0.5, 0.5])
+    roi2 = pg.RectROI([0.6, 0.6], [0.7, 0.7])
+    roi3 = pg.RectROI([0.8, 0.8], [0.9, 0.9])
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi1, roi2, roi3]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 3
+
+    plot_widget.remove_all_rois()
+
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_keyboard_bindings(qtbot, plot_widget, prepare_disc_test_data):
+    """Test changing coordinates and removing ROI."""
+    roi1 = pg.RectROI([0, 0], [0.5, 0.5])
+    roi2 = pg.RectROI([0.6, 0.6], [0.7, 0.7])
+    roi3 = pg.RectROI([0.8, 0.8], [0.9, 0.9])
+    plot_widget._onClick(*prepare_disc_test_data)
+    plot_widget.roi_list = [roi1, roi2, roi3]
+    plot_widget.plot()
+
+    assert len(plot_widget.roi_list) == 3
+
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_E, QtCore.Qt.ShiftModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 3
+
+    roi2.mouseHovering = True
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_D, QtCore.Qt.NoModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 2
+
+    event = QtGui.QKeyEvent(QtCore.QEvent.KeyPress, QtCore.Qt.Key_D, QtCore.Qt.ShiftModifier)
+    QtWidgets.QApplication.sendEvent(plot_widget, event)
+    assert len(plot_widget.roi_list) == 0
+
+
+def test_selection_from_roi(plot_widget, prepare_disc_test_data):
+    """Test selection of points from roi."""
+    plot_widget._onClick(*prepare_disc_test_data)
+    roi = pg.RectROI([-0.1, -0.1], [1.2, 1.2])
+    plot_widget.roi_list = [roi]
+
+    boolean_vector = plot_widget.get_selection()
+
+    assert len(boolean_vector) == DATA_LEN
+    assert np.sum(boolean_vector) == DATA_LEN
+
 
-RNG = np.random.default_rng(seed=0)
+def test_auto_range_disc(plot_widget, prepare_disc_test_data):
```

Comment:
in general I'd look for occurrences of "disc" and "cont" in the code and turn them into "discrete" and "continuous"